### PR TITLE
Migrate ssz into monorepo

### PR DESCRIPTION
**Pull request by wemeetagain**
Merges ssz into lodestar (under `packages/ssz`), keeping ssz git history
TODO (after this PR?):
- Archive ssz-js repo
- Update any relevant links pointing to ssz repo
  - ssz specs
- Fix monorepo autogenerated documentation pages

### CODE REVIEW

**Comment by GregTheGreek** [comment]
wow thats cool! lol

**Comment by GregTheGreek** [comment]
perhaps update the travis file?

**Comment by GregTheGreek** [comment]
If I recall the old ssz has some custom ci scripts - are those irrelevant or did we standardize everything?

File: packages/ssz/scripts/release-js.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
```
Comment (mpetrunic) [inline packages/ssz/scripts/release-js.sh]: do we need this file with lerna?

File: yarn.lock [hunk 2]
```diff
@@ -24,7 +24,7 @@
   dependencies:
```
Comment (mpetrunic) [inline yarn.lock:24]: this file is also not needed

**Review by mpetrunic — CHANGES_REQUESTED** [review]
You might wanna consider to use lodestar as npm dependency and move spec tests to root in this PR.

It should be fairly simple to do and CI won't have to pull spec tests twice :)

**Comment by wemeetagain** [comment]
> You might wanna consider to use lodestar as npm dependency and move spec tests to root in this PR.
> 
> It should be fairly simple to do and CI won't have to pull spec tests twice :)

I was hoping to make a spec-tests private package, where all the spec tests for lodestar, ssz, bls, etc would live. It would include the eth2.0-spec-tests-util, lodestar, ssz, bls as dependencies and have the eth2.0-spec-tests as a submodule within it. We would be able to remove the `test:spec` command from lodestar, ssz, etc, and just run all the spec tests from this package. This would also alleviate the circular dependencies in ssz required by spec tests. (ssz -> eth2.0-ssz-types -> ssz)

I thought it might be a little too big of a change for this PR, though, so I just left it as is.

File: packages/ssz/scripts/release-js.sh (hunk 1 above)
Comment (wemeetagain) [reply packages/ssz/scripts/release-js.sh]: no

File: yarn.lock (hunk 2 above)
Comment (wemeetagain) [reply yarn.lock:24]: I don't think I added this file as part of the PR.
I thought that the root-level yarn.lock is used by lerna/yarn workspaces. (as opposed to having a yarn.lock in every package)

Comment (mpetrunic) [reply yarn.lock:24]: oh yeah, it was werid, it shown that lock inside package, nvm this comment

File: packages/ssz/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# ssz
+[![Build Status](https://travis-ci.com/ChainSafe/ssz-js.svg?branch=master)](https://travis-ci.com/ChainSafe/ssz-js)
```
Comment (mpetrunic) [inline packages/ssz/README.md]: switch to lodestar badge?

**Review by mpetrunic — CHANGES_REQUESTED** [review]

**Review by mpetrunic — APPROVED** [review]

**Comment by wemeetagain** [comment]
some tests "failing" bc of nyc breaking after the fact, NOT due to failing tests.
Hopefully @mpetrunic 's recent PR will avoid this in the future

**Review by Mikerah — APPROVED** [review]
Looks good to me!
One thing I would generally comment on is more descriptive error messages. 
We can create a module just of lodestar specific error messages (now that ssz is in this repo).

**Pull request merged**